### PR TITLE
89385 Update Page Titles (In person appointments including COVID-19 vaccine and Claim Exam Appointment - Part 2)

### PR DESCRIPTION
**Pull request by vbahinwillit**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**: Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
- _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
This PR updates the title that displays in the browser tab.

- _(Summarize the changes that have been made to the platform)_
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#89385

## Testing done
Unit and e2e testing.

## Screenshots
**NOTE**: COVID appointments use the In-person layout.
Screenshot 2024-08-09 at 10 09 26 AM
Screenshot 2024-08-09 at 10 09 41 AM
Screenshot 2024-08-09 at 10 10 38 AM

---
Screenshot 2024-08-09 at 9 45 15 AM
Screenshot 2024-08-09 at 9 45 47 AM
Screenshot 2024-08-09 at 9 51 02 AM

## What areas of the site does it impact?
VAOS

## Acceptance criteria
- [ ] Given a user is on a `Confirmed` `In-person` details page
Then the browser tab title should read `In-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Upcoming` `In-person` details page
Then the browser tab title should read `In-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Past` `In-person` details page
Then the browser tab title should read `Past in-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Canceled` `In-person` details page
Then the browser tab title should read `Canceled in-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

---

- [ ] Given a user is on a `Confirmed` `COVID vaccine` details page
Then the browser tab title should read `In-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Upcoming` `COVID vaccine` details page
Then the browser tab title should read `In-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Past` `COVID vaccine` details page
Then the browser tab title should read `Past in-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Canceled` `COVID vaccine` details page
Then the browser tab title should read `Canceled in-person appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

### Claim exam appointments

- [ ] Given a user is on a `Upcoming` `Claim exam` details page
Then the browser tab title should read `Claim exam appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Past` `Claim exam` details page
Then the browser tab title should read `Past claim exam appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

- [ ] Given a user is on a `Canceled` `Claim exam` details page
Then the browser tab title should read `Canceled claim exam appointment on [Weekday], [Month] [DD], [YYYY] | Veterans Affairs`

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx [hunk 1]
```diff
@@ -60,18 +69,42 @@ export default function ConfirmedAppointmentDetailsPage() {
...
+        pageTitle = prefix ? `${prefix} phone` : 'Phone';
+      }
+
       const pageTitleSuffix = featureBreadcrumbUrlUpdate
         ? ' | Veterans Affairs'
         : '';
       if (appointment && appointmentDate) {
         document.title = `${pageTitle} appointment on ${appointmentDate.format(
-          'dddd, MMMM D, YYYY',
+          'dddd, MMMM DD, YYYY',
```
Comment (JunTaoLuo) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:94]: Hmm can we double check with design that this is what we actually want? I know in the specs it says `DD` but personally I find "August 09" to be odd. I like "August 9" better myself but just want to make sure we're all on the same page here.

cc @outerpress

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx [hunk 2]
```diff
@@ -60,18 +69,42 @@ export default function ConfirmedAppointmentDetailsPage() {
...
     },
-    // [appointment, appointmentDate, isCommunityCare, featureBreadcrumbUrlUpdateæ],
+    [
+      appointment,
+      appointmentDate,
+      isCommunityCare,
+      isCanceled,
+      isInPerson,
+      isPast,
+      featureBreadcrumbUrlUpdate,
```
Comment (JunTaoLuo) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:106]: Hmm linter is complaining that `isVideo` is missing from this list.

**Review by JunTaoLuo — DISMISSED** [review]
Changes look good, just a couple of small nits.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:106]: Done.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/index.jsx:94]: Checked with Peter and he agrees. No leading zero.

"No preceding zero pls. I think in this context it's easier to read without it"

cc @outerpress

**Review by JunTaoLuo — APPROVED** [review]
LGTM

**Pull request merged**